File: modules/agent/src/models/Reasoning.py
from sqlalchemy import Column, Text, DateTime, BigInteger, UniqueConstraint, func
from sqlalchemy.dialects.postgresql import JSONB
from models import Base


class Reasoning(Base):
    """Schema registry for AI agents - maps reasoning contexts to database tables."""

    __tablename__ = "Reasoning"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    type = Column(Text, nullable=False)  # 'crm', 'finance', etc.
    table_name = Column(Text, nullable=False)
    description = Column(Text, nullable=True)
    schema_hint = Column(JSONB, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

    __table_args__ = (
        UniqueConstraint("type", "table_name", name="uq_reasoning_type_table"),
    )

File: modules/agent/internal/models/Reasoning.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type Reasoning struct {
	ID          int64          `gorm:"primaryKey;autoIncrement" json:"id"`
	Type        string         `gorm:"type:text;not null" json:"type"` // crm, finance, etc.
	TableName_  string         `gorm:"column:table_name;type:text;not null" json:"table_name"`
	Description *string        `gorm:"type:text" json:"description"`
	SchemaHint  datatypes.JSON `gorm:"type:jsonb" json:"schema_hint"`
	CreatedAt   time.Time      `gorm:"autoCreateTime" json:"created_at"`
}

func (Reasoning) TableName() string {
	return "Reasoning"
}
